options/m_option.c

```diff
@@ -190,17 +190,18 @@ static void add_bool(const m_option_t *opt, void *val, double add, bool wrap)
 
 static int bool_set(const m_option_t *opt, void *dst, struct mpv_node *src)
 {
-    if (src->format != MPV_FORMAT_FLAG)
+    // MPV_FORMAT_FLAG is for libmpv compatibility
+    if (src->format != MPV_FORMAT_BOOL && src->format != MPV_FORMAT_FLAG)
         return M_OPT_UNKNOWN;
-    VAL(dst) = !!src->u.flag;
+    VAL(dst) = src->u.bool_;
```

Comment:
Again referring to the python script, reading a int 1 as a bool can result in false depending on endianness.

Getting rid of `m_option_type_flag` and treating `MPV_FORMAT_FLAG` as `CONF_TYPE_BOOL` might be a bad idea in general because of the difference in m_option_type_t.size causing problems with memcpy. Maybe it's already handled appropriately, I haven't checked all the places where this might matter, but since this is can lead to easy to miss problems, it should at least be a separate change to the introduction of `MPV_FORMAT_BOOL`, if at all.

Tbh I'd be fine with keeping `m_option_type_flag` around until `MPV_FORMAT_FLAG` gets actually dropped to avoid any potential problems.

Reply:
>it should at least be a separate change to the introduction of MPV_FORMAT_BOOL, if at all.

Yeah I can do that. It will look a little redundant because the same spots are going to be changed twice in a row, but probably for the best (at least to help decide if we're even going to accept this PR in any form at all).